pkg/controller/volume/expand/expand_controller.go

```diff
@@ -123,107 +113,156 @@ func NewExpandController(
 	expc.recorder = eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "volume_expand"})
 	blkutil := volumepathhandler.NewBlockVolumePathHandler()
 
-	expc.opExecutor = operationexecutor.NewOperationExecutor(operationexecutor.NewOperationGenerator(
+	expc.operationGenerator = operationexecutor.NewOperationGenerator(
 		kubeClient,
 		&expc.volumePluginMgr,
 		expc.recorder,
 		false,
-		blkutil))
-
-	expc.resizeMap = cache.NewVolumeResizeMap(expc.kubeClient)
+		blkutil)
 
 	pvcInformer.Informer().AddEventHandler(kcache.ResourceEventHandlerFuncs{
-		UpdateFunc: expc.pvcUpdate,
-		DeleteFunc: expc.deletePVC,
+		AddFunc: expc.enqueuePVC,
+		UpdateFunc: func(old, new interface{}) {
+			oldPVC, ok := old.(*v1.PersistentVolumeClaim)
+			if !ok {
+				return
+			}
+
+			oldSize := oldPVC.Spec.Resources.Requests[v1.ResourceStorage]
+			newPVC, ok := new.(*v1.PersistentVolumeClaim)
```

Comment:
It seems this (including the check on line 133) should be done before obtaining oldSize.